From 377fe91b0e72d585629b8711097f0491ecac4fab Mon Sep 17 00:00:00 2001
From: Yufei Gu <yufei.gu@cloudera.com>
Date: Fri, 10 Feb 2017 10:17:52 -0800
Subject: [PATCH 2202/2848] YARN-6112. UpdateCallDuration is calculated only
 when debug logging is enabled. (Yufei Gu via
 kasha)

(cherry picked from commit 41361ec39e84ec9949285b86ed97888ed4871bc4)

Change-Id: I9ffc302e4103469e3ef9fadbc538510f15c2d94e
---
 .../scheduler/fair/FairScheduler.java              |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairScheduler.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairScheduler.java
index 1e5aaa1..08887da 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairScheduler.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FairScheduler.java
@@ -289,6 +289,9 @@ public void run() {
           long start = getClock().getTime();
           update();
           long duration = getClock().getTime() - start;
+          // UpdateCall duration and ThreadRun Duration are the same after
+          // YARN-4752 (preemption overhaul), we keep both for compatibility.
+          fsOpDurations.addUpdateCallDuration(duration);
           fsOpDurations.addUpdateThreadRunDuration(duration);
         } catch (InterruptedException ie) {
           LOG.warn("Update thread interrupted. Exiting.");
@@ -326,8 +329,6 @@ public void run() {
    * required resources per job.
    */
   protected synchronized void update() {
-    long start = getClock().getTime();
-
     FSQueue rootQueue = queueMgr.getRootQueue();
 
     // Recursively update demands for all queues
@@ -351,9 +352,6 @@ protected synchronized void update() {
             "  Demand: " + rootQueue.getDemand());
       }
     }
-
-    long duration = getClock().getTime() - start;
-    fsOpDurations.addUpdateCallDuration(duration);
   }
 
   public synchronized RMContainerTokenSecretManager
-- 
1.7.9.5

